app/food_truck.rb

```diff
@@ -0,0 +1,26 @@
+# Retrieves food-truck feed and parses it
+
+require_relative '../constants'
+require 'httparty'
+require 'nokogiri'
+
+module FoodTruck
+  class FoodTruck
+    attr_reader :trucks
+    def initialize(url: FOOD_TRUCK_URL)
+      load_trucks_from(url)
+    end
+
+    private
+
+    def load_trucks_from(url)
+      request = HTTParty.get(url)
```

Comment:
The fact that you use `HTTParty` here to get the the schedule page means this class has two distinct responsibilities:

* Fetching the schedule page.
* Parsing the schedule page.

Abstractly, this is a violation of the [Single Responsibility Principle](https://blog.8thlight.com/uncle-bob/2014/05/08/SingleReponsibilityPrinciple.html). Concretely, it has some adverse effects:

* Tests for your parsing logic require an internet connection and are made slower by the need to fetch the markup they parse over the internet.
* Tests for your parsing logic are non-deterministic and unreliable.  They can fail any time due to a network problem or due to the remote server being down (for maintenance or whatever).  It's not good if you have tests that can randomly fail due to something you have no control over and that has nothing to do with the correctness of your logic here.
* Due to the non-determinism, your tests for this class make have very weak expectations: just that a non-empty array is returned.
* If there's some edge case bug that is occasionally hit in the parsing logic, there is no simple way to write a test for it because you can't deterministically feed it the markup that triggers the edge case in your test.
* If/when you add more robust logic around fetching the page (e.g. to handle transient network failures, or to handle redirects, or to use a caching proxy, or whatever) it's going to be entangled with your parsing logic here.

Instead, I would recommend you split this into two distinct classes:

* `FoodTruck::ScheduleFetcher`: responsible for fetching the schedule over the network.  (The HTTParty bits would go here, and this would ideally return a string of HTML markup).
* `FoodTruck::SchedulerParser`: responsible for parsing the schedule.  (This would receive the string of HTML markup a input).

This separation solves the problems above:

* It's easy to properly unit test this class with deterministic, robust tests: just create example markup strings and feed it in.
* Your expectations in those unit tests can be very string since the tests will be deterministic.
* Writing tests to cover edge cases that crop up becomes easy: just write a test that passes the problematic markup.
* You have distinct, non-entangled places to evolve the fetching and parsing logic individually.

If you go that route, you could either create a new class to encapsulate combining the fetcher and parser (basically, taking the place of this class) or you could combine them in the sinatra app.  I'd lean towards the latter -- the Sinatra app can coordinate these two things just fine, with no need for another layer of indirection, IMO.